design/k8s_secrets_labels_design.md

```diff
@@ -0,0 +1,218 @@
+# K8s Secrets Labels Design
+
+## Table of Contents
+
+[//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
+
+## Glossary
+
+| **Term**                                                     | **Description**                                              |
+| ------------------------------------------------------------ | ------------------------------------------------------------ |
+| [Label](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) | Key/value pairs that are attached to K8s objects. They are used to identify and group objects for effective querying and selection. |
+
+## Background
+
+As part of Phase 2 for Secrets Provider for K8s effort, we want to improve the customer experience by offering the ability to use labels for easy identification of K8s secrets whose secret values should come from Conjur. This will allow customers to list in terms of groupings instead of having to list the K8s secrets individually.
+
+## Issue description
+
+At current, the Secrets Provider code looks at the `K8S_SECRETS` environment variable in each of the pod manifests to know which K8s Secret to update with Conjur values. 
+
+Listing each K8s Secret in this way is not a scalable solution, especially for customers with hundreds of K8s Secrets. 
```

Comment:
why?

Reply:
As in why is it not scalable? It is not a scalable solution because customers will need to detail each secret that their apps need. So if a customer has 100 applications and each pod needs 2 secrets that is a lot of manual work

Reply:
but what's the alternative? In the scenario you describe, the customer will need to create 50 labels and define them across the k8s secrets manifests. Is that what you mean? Or do you mean that they will have only one label and each secerts-provider will retrieve all of them? 

Can you please describe what you expect the customer to do in the new feature in case of the use-case that you described above?